src/aws_request.erl

```diff
@@ -79,11 +79,12 @@ build_custom_headers(ParamsCustomHeadersMapping, Params0)
 
 %% @doc Add querystring to url is there are any parameters in the list
 -spec add_query(binary(), [{binary(), any()}]) -> binary().
-add_query(Url, []) ->
-    Url;
-add_query(Url, Query) ->
-    QueryString = aws_util:encode_query(Query),
-    aws_util:binary_join([Url, QueryString], <<"?">>).
+add_query(Url0, Query0) ->
+  [Url | _] = string:split(Url0, <<"?">>),
+  HackneyUrl = hackney_url:parse_url(Url0),
```

Comment:
hackney_url:parse_url/1 is used to get the query string, but the not the URL. Would something like this work?

```erlang
HackneyUrl = hackney_url:parse_url(Url0),
NewQs = aws_util:encode_query(hackney_url:parse_qs(HackneyUrl#hackney_url.qs) ++ Query0)),
HackneyUrlWithAddedQs = HackneyUrl#hackney_url{qs = NewQs},
hackney_url:unparse_url(HackneyUrlWithAddedQs).
```